### ISSUE TITLE
js-mode not working on jsx file

### ISSUE THREAD

**Issue by Petricu** [hlissner/doom-emacs#3569]
**What did you expect to happen?**
...

When opening a React project(with projectile), should get completion and linting.

**What actually happened?**
...
Nothing. Syntax highlighting is ok, but no completion, no linting. By manually enabling js-jsx-mode I get some snippets, but nothing project aware

**Additional details:**
- Include a link to your private config
- Include screenshots/casts of your issue
- If you mention key sequences, include what commands they're bound to (use `SPC
  h k KEY` or `C-h h k KEY` to inspect keys).
- Include any warnings or errors logged to \*Messages\* (use `M-x
  view-echo-area-messages` to see it).

<details><pre>
If an error message is involved include a backtrace of it.

How to acquire a backtrace:
https://github.com/hlissner/doom-emacs/blob/develop/docs/getting_started.org#how-to-extract-a-backtrace-from-an-error
</pre></details>

**Steps to reproduce:**
1. Select these example steps,
2. Delete them,
3. And replace them with precise steps to reproduce your issue.
4. Fill in "system information" below.

**System information:**
<details><pre>
((emacs
  (version . "27.0.91")
  (features . "XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD JSON PDUMPER LCMS2 GMP")
  (build . "iul 15, 2020")
  (buildopts "--prefix= --prefix=/snap/emacs/current/usr --with-xwidgets --with-x-toolkit=gtk3 --without-xaw3d --with-modules --with-cairo 'CFLAGS= -isystem/build/emacs/parts/emacs/install/usr/include' 'CPPFLAGS= -isystem/build/emacs/parts/emacs/install/usr/include' 'LDFLAGS= -L/build/emacs/parts/emacs/install/lib -L/build/emacs/parts/emacs/install/usr/lib -L/build/emacs/parts/emacs/install/lib/x86_64-linux-gnu -L/build/emacs/parts/emacs/install/usr/lib/x86_64-linux-gnu'")
  (windowsys . batch)
  (daemonp))
 (doom
  (version . "2.0.9")
  (build . "grafted, HEAD -> develop, origin/develop, origin/HEAD 1cec7bf 2020-07-12 22:48:47 -0400")
  (dir . "~/.doom.d/"))
 (system
  (type . gnu/linux)
  (config . "x86_64-pc-linux-gnu")
  (shell . "/bin/zsh")
  (uname . "Linux 5.4.0-40-generic #44-Ubuntu SMP Tue Jun 23 00:01:04 UTC 2020 x86_64")
  (path "~/.cargo/bin" "/usr/local/sbin" "/usr/local/bin" "/usr/sbin" "/usr/bin" "/sbin" "/bin" "/usr/games" "/usr/local/games" "/snap/bin" "/snap/emacs/current/usr/libexec/emacs/27.0.91/x86_64-pc-linux-gnu"))
 (config
  (envfile . envvar-file)
  (elc-files . 0)
  (modules :completion company (ivy +fuzzy) :ui doom doom-dashboard doom-quit hl-todo indent-guides minimap modeline ophints (popup +defaults) unicode vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) file-templates fold (format +onsave) multiple-cursors snippets word-wrap :emacs dired electric undo vc :term eshell :checkers syntax spell grammar :tools (debugger +lsp) (docker +lsp) editorconfig ein (eval +overlay) gist lookup lsp magit make pdf prodigy taskrunner :lang emacs-lisp (json +lsp) (javascript +lsp) org (rust +lsp) (sh +lsp) (web +lsp) :config (default +bindings +smartparens))
  (packages (prettier))
  (unpin "n/a")
  (elpa "n/a")))

>         
</pre></details>

**Comment by hlissner** [comment]
I see you have `:lang (javascript +lsp)` enabled. 

1. What LSP server(s) do you have installed? 
2. Is it connecting/initializing correctly when you open a js2-mode (we use js2-mode, not js-mode, btw) buffer?
3. Any warnings/errors in the \*Messages\* buffer when opening a js2-mode buffer, or trying to code-complete there?
4. Does that LSP server's project page claim that it can do the kind of code completion you want?

**Comment by Petricu** [comment]
Figured it out, actually it doesn't start the lsp on buffer open. It is fixable by manually starting it with M-x lsp. The lsp I use is the one recommended by the doom docs, think it's typescript-language-server. Sorry for misinforming, I'm a rookie at JS within Emacs

**Comment by hlissner** [comment]
That's very strange. If you have `:lang (javascript +lsp)` enabled (and you do), it should start lsp-mode for you. This is its behavior in vanilla Doom at least.

In fact, js files should open in `js2-mode` and jsx files should open in `rjsx-mode`, not `js-mode`. Have you overridden the `.js` entry in `auto-mode-alist` from your private config? What major mode are `.jsx` files being opened in? (`SPC h v major-mode` will tell you the value of the `major-mode` variable in the current buffer)

**Comment by Petricu** [comment]
Yes, it opens in rjsx-mode on jsx files and js2-mode on js files, as expected. The problem is that lsp is not automatically started on those files. Manually starting it fixes it. I changed the issue title, so as if anyone else has this problem they could hop in
